iroha/src/block.rs

```diff
@@ -218,6 +237,51 @@ impl ValidBlock {
     pub fn verified_signatures(&self) -> Vec<Signature> {
         self.signatures.verified(self.hash().as_ref())
     }
+
+    /// Checks if there are no transactions in this block.
+    pub fn is_empty(&self) -> bool {
+        self.transactions.is_empty() && self.rejected_transactions.is_empty()
+    }
+}
+
+impl From<&ValidBlock> for Vec<Event> {
+    fn from(block: &ValidBlock) -> Self {
+        block
+            .transactions
+            .iter()
+            .cloned()
+            .map(|transaction| {
+                PipelineEvent {
+                    entity_type: PipelineEntityType::Transaction,
+                    status: PipelineStatus::Validating,
+                    hash: transaction.hash(),
+                }
```

Comment:
`PipelineEvent::validating_transaction(hash: Hash)` or `PipelineEvent::new(EntityType::Transaction, Status::Validating, hash)`.

Reply:
Yes, I think simply adding `PipelineEvent::new(..)` here would be better. Also relates to my previous comments above.